ironfish/src/rpc/adapters/httpAdapter.ts

```diff
@@ -164,86 +167,123 @@ export class RpcHttpAdapter implements IRpcAdapter {
+          // TODO: Most HTTP clients don't parse `Transfer-Encoding: chunked` by chunk
+          // they wait until all chunks have been received and combine them. This will
+          // stream a delimitated list of JSON objects but is still probably not
+          // ideal as a response. We could find some better way to stream
+          const delimeter = chunkStreamed ? MESSAGE_DELIMITER : ''
+
+          const responseData = JSON.stringify({ data })
+          const responseSize = Buffer.byteLength(responseData, 'utf-8')
+          this.outboundTraffic.add(responseSize)
+          this.outboundBytes.value += responseSize
+
+          response.write(delimeter + responseData)
+          chunkStreamed = true
+        },
+      )
+
+      const currRequest = this.requests.get(requestId)
+      currRequest && this.requests.set(requestId, { ...currRequest, rpcRequest })
+
+      await router.route(route, rpcRequest)
+    } catch (e) {
+      const error = ErrorUtils.renderError(e)
+      this.logger.debug(`Error in HTTP adapter: ${error}`)
+
+      const responseMessage: RpcHttpError =
+        e instanceof RpcResponseError
+          ? {
+              code: e.code,
+              status: e.status,
+              message: e.message,
+              stack: e.stack,
+            }
+          : {
+              code: ERROR_CODES.ERROR,
+              status: 500,
+              message: error,
+            }
 
-        response.end(delimeter + responseData)
+      // If we sent a streaming response we cannot send
+      // headers again with the status
+      if (!response.headersSent) {
+        response.writeHead(responseMessage.status)
+      }
 
-        this.cleanUpRequest(requestId)
-      },
-      (data: unknown) => {
-        // TODO: Most HTTP clients don't parse `Transfer-Encoding: chunked` by chunk
-        // they wait until all chunks have been received and combine them. This will
-        // stream a delimitated list of JSON objects but is still probably not
-        // ideal as a response. We could find some better way to stream
-        const delimeter = chunkStreamed ? MESSAGE_DELIMITER : ''
-
-        const responseData = JSON.stringify({ data })
-        const responseSize = Buffer.byteLength(responseData, 'utf-8')
-        this.outboundTraffic.add(responseSize)
-        this.outboundBytes.value += responseSize
-
-        response.write(delimeter + responseData)
-        chunkStreamed = true
-      },
-    )
+      const delimeter = chunkStreamed ? MESSAGE_DELIMITER : ''
 
-    const currRequest = this.requests.get(requestId)
-    currRequest && this.requests.set(requestId, { ...currRequest, rpcRequest })
+      const responseData = JSON.stringify(responseMessage)
+      const responseSize = Buffer.byteLength(responseData, 'utf-8')
+      this.outboundTraffic.add(responseSize)
+      this.outboundBytes.value += responseSize
 
-    await router.route(route, rpcRequest)
+      response.end(delimeter + responseData)
```

Comment:
Fixed not sending a delimeter when sending an error if a chunk has been streamed which would crash the client.

Reply:
nit: delimiter, not delimeter

Reply:
https://github.com/iron-fish/ironfish/pull/4579

Reply:
Fixed